graphql_compiler/compiler/sqlalchemy_extensions.py

```diff
@@ -64,15 +66,31 @@ def print_sqlalchemy_query_string(query, dialect):
     Returns:
         string that can be ran using sqlalchemy.sql.text(result)
     """
+    supported_dialects = (PGDialect_psycopg2, MSDialect_pyodbc)
+    if isinstance(dialect, supported_dialects):
+        class BindparamCompiler(dialect.statement_compiler):
+            def visit_bindparam(self, bindparam, **kwargs):
+                # A bound parameter with name param is represented as ":param". However,
+                # if the parameter is expanding (list-valued) it is represented as
+                # "([EXPANDING_param])" by default. This is an internal sqlalchemy
+                # representation that is not understood by databases, so we explicitly
+                # make sure to print it as ":param".
+                return f":{bindparam.key}"
+
+        return str(BindparamCompiler(dialect, query).process(query))
+    else:
+        raise AssertionError(f"Unsupported dialect {dialect}. "
+                             f"Only {supported_dialects} are supported.")
+
+
+def bind_parameters_to_query_string(query, parameters):
+    bound_parameters = [
+        sqlalchemy.bindparam(
+            parameter_name,
+            value=parameter_value,
+            expanding=isinstance(parameter_value, (list, tuple)),
```

Comment:
Is it feasible to read this information from the parameter metadata instead? The metadata is the canonical place for this info, and I'd hate to see us add support for some new collection type (e.g. `np.array()`) as a parameter input type only to create the wrong `bindparam` here. It feels like it might be painful bug to track down.

Reply:
The behaviour of `np.array()` should be the same. I get the sentiment though, and it would be good. I'd like to resolve the `graphql_to_sql` top-level-function comment first to avoid plumbing twice.